### PR TITLE
Update studyspacesUITests.swift

### PR DESCRIPTION
**Pull request by garfieldmaitland**
## Ticket Number

[SCRUM-21](https://we-dev-backend.atlassian.net/jira/software/projects/SCRUM/boards/1/timeline?selectedIssue=SCRUM-21)

## Context

Additional setup of UI test suite for the study spaces app. These tests will help ensure core functionality remains stable as we develop the app.

## Purpose

- Establish baseline UI testing infrastructure
- Ensure critical user flows are tested
- Support future feature development with test coverage

## Description

Added 9 comprehensive UI test suite including:
- Basic app launch and performance tests
- Tab bar navigation tests
- Search functionality verification
- Filter sheet interaction tests
- Map interaction tests
- Space detail view tests
- Networking connection tests
- Location permission handling tests
- Error state verification

### CODE REVIEW

File: studyspacesUITests/studyspacesUITests.swift [hunk 1]
```diff
@@ -38,4 +38,123 @@ final class studyspacesUITests: XCTestCase {
...
     }
+
+    func testBasicTabBarExists() throws {
+        let app = XCUIApplication()
+        app.launch()
+        
+        // Test basic UI elements exist
+        XCTAssertTrue(app.tabBars.firstMatch.exists, "Tab bar should exist")
+        XCTAssertTrue(app.searchFields.firstMatch.exists, "Search field should exist")
+        XCTAssertTrue(app.buttons["FilterButton"].exists, "Filter button should exist")
```
Comment (DilanRocco) [inline studyspacesUITests/studyspacesUITests.swift:49]: How do you plan to enforce a 'FilterButton' will exist in the future ?

Comment (garfieldmaitland) [reply studyspacesUITests/studyspacesUITests.swift:49]: If not, we can remove/comment it out

Comment (DilanRocco) [reply studyspacesUITests/studyspacesUITests.swift:49]: There's a lot of these types of statements in this code.

Comment (garfieldmaitland) [reply studyspacesUITests/studyspacesUITests.swift:49]: Removed or commented out?

Comment (DilanRocco) [reply studyspacesUITests/studyspacesUITests.swift:49]: I think these should be added after the code has been developed

Comment (garfieldmaitland) [reply studyspacesUITests/studyspacesUITests.swift:49]: I appreciate your feedback regarding the unit tests. In Test-Driven Development (TDD), writing tests before implementing code is a standard practice to ensure functionality and guide development. For reference, the book ['Test-Driven iOS Development with Swift - Fourth Edition'](https://github.com/PacktPublishing/Test-Driven-iOS-Development-with-Swift-Fourth-Edition?tab=readme-ov-file) emphasizes this approach. Could we discuss any specific concerns you have about including these tests now? I'm open to adjusting our process to align with team standards

**Comment by garfieldmaitland** [comment]
I've addressed your feedback on the unit tests and clarified their role in TDD. Please, let me know if there is anything further to specify

File: studyspacesUITests/studyspacesUITests.swift (hunk 1 above)
Comment (DilanRocco) [reply studyspacesUITests/studyspacesUITests.swift:49]: There are use cases when the writing tests before you code is written. However, the tests in this PR makes a lot of unnecessary assumptions. For example, if we were building the function for addition, we could make tests like (assert 3 + 3 = 6) or (assert 4 + 10 = 14) (using pseudocode) and if those tests are incorrect, our add method is incorrect. However, with the tests in this PR, there are a lot of assumptions about names of buttons that may or may not exist. When this code eventually gets written, the tabBar button 'Saved' might not exist. app.tables["SearchResults"] may not exist. 

In general, UI tests are not done before hand and very often are autogenerated.

**Comment by DilanRocco** [comment]
> I've addressed your feedback on the unit tests and clarified their role in TDD. Please, let me know if there is anything further to specify

I added a new comment

File: studyspacesUITests/studyspacesUITests.swift (hunk 1 above)
Comment (garfieldmaitland) [reply studyspacesUITests/studyspacesUITests.swift:49]: Understood. Message received

**Pull request closed without merging**